pkg/ebpf/c/maps.h

```diff
@@ -78,17 +97,30 @@ enum tail_call_id_e
-BPF_HASH(mnt_ns_filter, u64, eq_t, 256);                           // filter events by mount namespace id
-BPF_HASH(pid_ns_filter, u64, eq_t, 256);                           // filter events by pid namespace id
-BPF_HASH(uts_ns_filter, string_filter_t, eq_t, 256);               // filter events by uts namespace name
-BPF_HASH(comm_filter, string_filter_t, eq_t, 256);                 // filter events by command name
-BPF_HASH(cgroup_id_filter, u32, eq_t, 256);                        // filter events by cgroup id
-BPF_HASH(binary_filter, binary_t, eq_t, 256);                      // filter events by binary path and mount namespace
-BPF_HASH(events_map, u32, event_config_t, MAX_EVENT_ID);           // map to persist event configuration data
+
+// versioned maps
+BPF_HASH_INNER(uid_filter, u32, eq_t, 256);                        // filter events by UID prototype, for specific UIDs either by == or !=
+BPF_HASH_OUTER(uid_filter_version, uid_filter, 64);                // map of UID filters maps
+BPF_HASH_INNER(pid_filter, u32, eq_t, 256);                        // filter events by PID prototype
+BPF_HASH_OUTER(pid_filter_version, pid_filter, 64);                // map of PID filters maps
+BPF_HASH_INNER(mnt_ns_filter, u64, eq_t, 256);                     // filter events by mount namespace id prototype
+BPF_HASH_OUTER(mnt_ns_filter_version, mnt_ns_filter, 64);          // map of mount namespace filters maps
+BPF_HASH_INNER(pid_ns_filter, u64, eq_t, 256);                     // filter events by pid namespace id prototype
+BPF_HASH_OUTER(pid_ns_filter_version, pid_ns_filter, 64);          // map of pid namespace filters maps
+BPF_HASH_INNER(uts_ns_filter, string_filter_t, eq_t, 256);         // filter events by uts namespace name prototype
+BPF_HASH_OUTER(uts_ns_filter_version, uts_ns_filter, 64);          // map of uts namespace filters maps
+BPF_HASH_INNER(comm_filter, string_filter_t, eq_t, 256);           // filter events by command name prototype
+BPF_HASH_OUTER(comm_filter_version, comm_filter, 64);              // map of command name filters maps
+BPF_HASH_INNER(cgroup_id_filter, u32, eq_t, 256);                  // filter events by cgroup id prototype
+BPF_HASH_OUTER(cgroup_id_filter_version, cgroup_id_filter, 64);    // map of cgroup id filters maps
+BPF_HASH_INNER(binary_filter, binary_t, eq_t, 256);                // filter events by binary path and mount namespace prototype
+BPF_HASH_OUTER(binary_filter_version, binary_filter, 64);          // map of binary filters maps
+BPF_HASH_INNER(process_tree_map, u32, eq_t, 10240);                // filter events by the ancestry of the traced process
+BPF_HASH_OUTER(process_tree_map_version, process_tree_map, 64);    // map of process tree maps
+BPF_HASH_INNER(events_map, u32, event_config_t, MAX_EVENT_ID);     // map to persist event configuration data
+BPF_HASH_OUTER(events_map_version, events_map, 64);                // map of events maps
```

Comment:
It's according the new map creation/insertion. Rafael also noticed this size. It was put as 64 just to align the max number of policies, however, we can reduce it to 4-8. I assume it would be unlikely to have to keep more than that, even if we compute new policies very quickly.

Reply:
Anyway, it would keep up to 64 references to the respective filter map:

```
❯ sudo bpftool map show name events_map_vers
78: hash_of_maps  name events_map_vers  flags 0x0
        key 2B  value 4B  max_entries 64  memlock 9280B
        pids tracee(51042)
```

Changing max_entries to 2 only reduced memlock by ~53%. The usage is minimal.

```
❯ sudo bpftool map show name events_map_vers
188: hash_of_maps  name events_map_vers  flags 0x0
        key 2B  value 4B  max_entries 2  memlock 4320B
        pids tracee(53850)
```